core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -427,6 +449,15 @@ class BlockManagerMasterEndpoint(
       locations.remove(blockManagerId)
     }
 
+    if (storageLevel.useDisk && externalShuffleServiceEnabled) {
```

Comment:
have you done any tests on what happens when a block is cached as MEMORY_AND_DISK?  I know that the meaning of `storageLevel` is a little ambiguous and can change depending on where the msg is.  We should make sure you're not getting `useDisk = true` here is the block is only in memory currently.

Reply:
I think according to the code we are safe here as a few lines above we are calling `updateBlockInfo` with the `storageLevel`:

https://github.com/attilapiros/spark/blob/a25e9aaf67df519a61078637dfb91d87b5173862/core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala#L435-L437

And the same `storageLevel` is used to create the `BlockStatus` where only the `diskSize` is filled:

https://github.com/attilapiros/spark/blob/a25e9aaf67df519a61078637dfb91d87b5173862/core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala#L610-L612

Reply:
A bit weird that in the map we are overwriting the one which calculated for (useMemory=true):

https://github.com/attilapiros/spark/blob/a25e9aaf67df519a61078637dfb91d87b5173862/core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala#L596-L598

But this cleanups I plan to do that in separate PR to keep my current PR focused to the new feature and make the reviewers life easier.

Reply:
I checked how the the message is created and this is where the storageLevel is computed (called from `doPutIterator`):

https://github.com/attilapiros/spark/blob/a25e9aaf67df519a61078637dfb91d87b5173862/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L693